Mathlib/RingTheory/Flat/FaithfullyFlat/Algebra.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2025 Christian Merten, Yi Song, Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten, Yi Song, Sihan Su
+-/
+import Mathlib.RingTheory.Flat.FaithfullyFlat.Basic
+import Mathlib.RingTheory.Ideal.Over
+import Mathlib.RingTheory.LocalRing.RingHom.Basic
+import Mathlib.RingTheory.Spectrum.Prime.Basic
+import Mathlib.RingTheory.TensorProduct.Quotient
+
+/-!
+# Properties of faithfully flat algebras
+
+An `A`-algebra `B` is faithfully flat if `B` is faithfully flat as an `A`-module. In this
+file we give equivalent characterizations of faithful flatness in the algebra case.
+
+## Main results
+
+Let `B` be a faithfully flat `A`-algebra:
+
+- `Ideal.comap_map_eq_self_of_faithfullyFlat`: the contraction of the extension of any ideal of
+  `A` to `B` is the ideal itself.
+- `Module.FaithfullyFlat.tensorProduct_mk_injective`: The natural map `M →ₗ[A] B ⊗[A] M` is
+  injective for any `A`-module `M`.
+- `PrimeSpectrum.specComap_surjective_of_faithfullyFlat`: The map on prime spectra induced by
+  a faithfully flat ring map is surjective. See also
+  `Ideal.exists_isPrime_liesOver_of_faithfullyFlat` for a version stated in terms of
+  `Ideal.LiesOver`.
+
+Conversely, let `B` be a flat `A`-algebra:
+
+- `Module.FaithfullyFlat.of_specComap_surjective`: `B` is faithfully flat over `A`,
+  if the induced map on prime spectra is surjective.
+- `Module.FaithfullyFlat.of_flat_of_isLocalHom`: flat + local implies faithfully flat
+
+-/
+
+universe u v
+
+variable {A : Type u} {B : Type v} [CommRing A] [CommRing B] [Algebra A B]
+
+open TensorProduct LinearMap
+
+/-- If `B` is a faithfully flat `A`-module and `M` is any `A`-module, the canonical
```

Comment:
Can you create a section with `variable [Module.FaithfullyFlat A B]` rather than adding it to various lemmas? Or maybe move the results that require it at the end without a section, as you prefer.